### DISCUSSION TITLE
Disable GET/?location

### DISCUSSION THREAD

**Discussion by pradeep8985** [iterate-ch/cyberduck#14560] [Q&A]
Hello All,

I am connecting to an s3 compatible storage and when I first connect to the storage, I see a lag in logging in. Upon checking the logs, the response time to the "GET /?location" is taking time in our case, while I noticed some of the other tools are not validating this operation at all... So I was wondering if there is any way to disable this check in Cyberduck... Thank you in advance. 

```
GET /?location= HTTP/1.1
Date: Fri, 21 Apr 2023 14:50:04 GMT
uAuthorization: ********
Connection: Keep-Alive
User-Agent: Cyberduck/8.4.3.38269 (Windows 10/10.0) (amd64)
Accept-Encoding: gzip,deflate

HTTP/1.1 200 OK
Date: Fri, 21 Apr 2023 14:50:08 GMT
x-amz-request-id: *****************
Gateway-Protocol: s3
Content-Length: 107
```

**Comment by dkocher** [comment]
It cannot currently be disabled but we must review [^1] if it should only be attempted to read the location when connecting to a single bucket.

[^1]: https://github.com/iterate-ch/cyberduck/blob/313b56d64861a4e49ac5abec8fcf1a940ab58920/s3/src/main/java/ch/cyberduck/core/s3/S3Session.java#L226-L253

**Reply by pradeep8985** [reply]
> It cannot currently be disabled but we must review [1](#user-content-fn-1-a94d8f248227e5272684cfd2c3635c57) if it should only be attempted to read the location when connecting to a single bucket.
> 
> ## Footnotes
> 1. https://github.com/iterate-ch/cyberduck/blob/313b56d64861a4e49ac5abec8fcf1a940ab58920/s3/src/main/java/ch/cyberduck/core/s3/S3Session.java#L226-L253 [↩](#user-content-fnref-1-a94d8f248227e5272684cfd2c3635c57)

I have tried using the region key and value set to us-east-1 and I couldn't see the GET /?location request in the logs anymore...
